src/signatures/commitment_signature.rs

```diff
@@ -24,32 +24,40 @@ pub enum CommitmentSignatureError {
-///
-/// In short, a Commitment Signature is made up of the tuple _(R, u, v)_, where _R_ is a random Pedersen commitment (of
-/// two secret nonces) and _u_ and _v_ are the two publicly known private signature keys. It demonstrates ownership of
-/// a specific commitment.
-///
-/// The Commitment Signature signes a challenge with the value commitment's value and blinding factor. The two nonces
-/// should be completely random and never reused - that responsibility lies with the calling function.
-///   C = a*H + x*G          ... (Pedersen commitment to the value 'a' using blinding factor 'x')
-///   R = k_2*H + k_1*G      ... (a public (Pedersen) commitment nonce created with the two random nonces)
-///   u = k_1 + e.x          ... (the first publicly known private key of the signature signing with 'x')
-///   v = k_2 + e.a          ... (the second publicly known private key of the signature signing with 'a')
-///   signature = (R, u, v)  ... (the final signature tuple)
-///
-/// Verification of the Commitment Signature (R, u, v) entails the following:
-///   S = v*H + u*G          ... (Pedersen commitment of the publicly known private signature keys)
-///   S =? R + e.C           ... (final verification)
-#[allow(non_snake_case)]
+/// # Commitment signatures
+/// 
+/// Given a commitment `commitment = a*H + x*G` and group element `pubkey = y*G`, a commitment signature is based on
+/// a representation proof of both openings: `(a, x)` and `y`. It additionally binds to arbitrary message data `m`
+/// via the challenge to produce a signature construction.
+/// 
+/// It is used in Tari protocols as part of transaction authorization.
+/// 
+/// The construction works as follows:
+/// - Sample scalar nonces `r_a, r_x, r_y` uniformly at random.
+/// - Compute ephemeral values `ephemeral_commitment = r_a*H + r_x*G` and `ephemeral_pubkey = r_y*G`.
+/// - Use strong Fiat-Shamir to produce a challenge `e`. If `e == 0` (this is unlikely), abort and start over.
+/// - Compute the responses `u_a = r_a + e*a` and `u_x = r_x + e*x` and `u_y = r_y + e*y`.
+/// 
+/// The signature is the tuple `(ephemeral_commitment, ephemeral_pubkey, u_a, u_x, u_y)`.
+/// 
+/// To verify:
+/// - The verifier computes the challenge `e` and rejects the signature if `e == 0` (this is unlikely).
+/// - Verification succeeds if and only if the following equations hold:
+///     `u_a*H + u*x*G == ephemeral_commitment + e*commitment`
+///     `u_y*G == ephemeral_pubkey + e*pubkey`
+/// 
+/// We note that it is possible to make verification slightly more efficient. To do so, the verifier selects a nonzero
+/// scalar weight `w` uniformly at random (not through Fiat-Shamir!) and accepts the signature if and only if the
+/// following equation holds:
+///     `u_a*H + (u_x + w*u_y)*G - ephemeral_commitment - w*ephemeral_pubkey - e*commitment - (w*e)*pubkey == 0`
+/// The use of efficient multiscalar multiplication algorithms may also be useful for efficiency.
```

Comment:
This seems like a good suggestion.

Reply:
I'll be implementing it in a subsequent commit! I want to do a few efficiency tests first.